src/api/comment/content-types/comment/lifecycles.js

```diff
@@ -19,23 +19,32 @@ module.exports = {
     const idea = comment.idea_card;
     const user = comment.user;
 
-    await strapi.entityService.create("api::subscription.subscription", {
-      data: {
-        entityType: "Comment",
-        entityId: commentId,
-        createdDateTime: new Date(),
-        active: true,
-        user: user,
-      },
-    });
+    /** verify that user isn't already subscribed to ideaCard */
+    const existingSubscriptions = await strapi.entityService.findMany(
+      "api::subscription.subscription",
+      {
+        filters: { user: user, entityId: idea.id },
+      }
+    );
+
+    if (!existingSubscriptions.length)
+      await strapi.entityService.create("api::subscription.subscription", {
+        data: {
+          entityType: "Comment",
+          entityId: idea.id,
+          createdDateTime: new Date(),
+          active: true,
+          user: user,
+        },
+      });
 
     strapi.entityService.create("api::event.event", {
       data: {
         action: "Commented",
         entityName: idea.ideaName,
         content: commentText,
         entityType: "Comment",
-        entityId: commentId,
+        entityId: idea.id,
```

Comment:
Can you add an integration test that would catch this error?

Reply:
okay

Reply:
Yes.